pkgs/development/misc/resholve/resholve-package.nix

```
@@ -0,0 +1,128 @@
+{ stdenv, lib, resholve }:
+
+{ pname
+, src
+, version
+, passthru ? { }
+  /* Each "solution" (k=v pair) describes one resholve invocation.
+   * See 'man resholve' for option details.
+   *
+   * Example:
+   * {
+   *   # name the script(s) for overrides & error messages
+   *   shortname = {
+   *     # required
+   *     # $out-relative paths to try resolving
+   *     scripts = [ "bin/shunit2" ];
+   *     # packages to resolve executables from
+   *     inputs = [ coreutils gnused gnugrep findutils ];
+   *     # path for shebang, or 'none' to omit shebang
+   *     interpreter = "${bash}/bin/bash";
+   *
+   *     # optional
+   *     # to translate fake/fix/keep directives, convert:
+   *     # - `key:value1;value2` to `key: [ value1 value2 ];`
+   *     # - `value` to `value = true;`
+   *     fake = { fake directives };
+   *     fix = { fix directives };
+   *     keep = { keep directives };
+   *     # file to inject before first code-line of script
+   *     prologue = file;
+   *     # file to inject after last code-line of script
+   *     epilogue = file;
+   *     # extra command-line flags passed to resholve; generally this API
+   *     # should align with what resholve supports, but flags may help if
+   *     # you need to override the version of resholve.
+   *     flags = [ ];
+   *   };
+   * }
+   */
+, solutions
+, ...
+}@attrs:
+let
+  inherit stdenv;
+  /* These functions break up the work of partially validating the
+   * 'solutions' attrset and massaging it into env/cli args.
+   *
+   * Note: some of the left-most args do not *have* to be passed as
+   * deep as they are, but I've done so to provide more error context
+   */
+
+  # for brevity / line length
+  spaces = l: builtins.concatStringsSep " " l;
+  semicolons = l: builtins.concatStringsSep ";" l;
+
+  /* Throw a fit with dotted attr path context */
+  nope = path: msg:
+    throw "${builtins.concatStringsSep "." path}: ${msg}";
+
+  /* Special-case directive value representations by type */
+  makeDirective = solution: env: name: val:
+    if builtins.isInt val then builtins.toString val
+    else if builtins.isString val then name
+    else if true == val then name
+    else if false == val then "" # omit!
+    else if null == val then "" # omit!
```

```suggestion
    else if false == val or null == val then "" # omit!
```

Not sure if this works in nix right now.

---

It does. Ex: https://github.com/NixOS/nixpkgs/blob/487141bc9328a24d96e916a6ca71dffdf4a4c743/lib/systems/default.nix#L33-L45